src/verify_cert.rs

```diff
@@ -517,28 +577,41 @@ impl ExtendedKeyUsage {
 
 /// An OID value indicating an Extended Key Usage (EKU) key purpose.
 #[derive(Clone, Copy)]
-struct KeyPurposeId {
-    oid_value: untrusted::Input<'static>,
+struct KeyPurposeId<'a> {
+    oid_value: untrusted::Input<'a>,
 }
 
-impl KeyPurposeId {
+impl<'a> KeyPurposeId<'a> {
     /// Construct a new [`KeyPurposeId`].
     ///
     /// `oid` is the OBJECT IDENTIFIER in bytes.
-    const fn new(oid: &'static [u8]) -> Self {
+    const fn new(oid: &'a [u8]) -> Self {
         Self {
             oid_value: untrusted::Input::from(oid),
         }
     }
 }
 
-impl PartialEq<Self> for KeyPurposeId {
+impl fmt::Debug for KeyPurposeId<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "KeyPurposeId(")?;
+        for (i, byte) in self.oid_value.as_slice_less_safe().iter().enumerate() {
```

Comment:
Not sure this is quite right. `oid_value` is the bytes of the encoding, but isn't the same as the normal dotted presentation of OIDs.

For example, in the tests below the OID [1.3.6.1.5.5.7.3.1](https://oid-base.com/get/1.3.6.1.5.5.7.3.1) is printed as `KeyPurposeId(43.6.1.5.5.7.3.2)`. That is because the first byte is 40 * first + second (see https://letsencrypt.org/docs/a-warm-welcome-to-asn1-and-der/#object-identifier-encoding for a better description). Aside from that this also is wrong for larger elements, which get split over several bytes.

Reply:
Doh... should be better now.